### PR TITLE
Feature/aa 401 add api for displaying actual hotel confirmation code and history

### PR DESCRIPTION
**Pull request by AlexeyShok**
Added endpoint to get actual hotel confirmation code and status for hotel confirmation page. Hotel confirmation code added to AccommodationBookingDetails for accept in admin panel (maybe in agent app too). Added endpoints to get hotel confirmation history for agents, admins and property owner employees.

### CODE REVIEW

File: Api/Controllers/AgentControllers/BookingsController.cs [hunk 1]
```diff
@@ -338,6 +338,23 @@ public async Task<IActionResult> GetBookingStatusHistory(int bookingId)
         }
 
 
+        /// <summary>
+        ///     Gets booking confirmation changes history
+        /// </summary>
+        /// <param name="referenceCode">Booking reference code for retrieving confirmation change history</param>
+        /// <returns>List of booking confirmation change events</returns>
+        [HttpGet("refcode/{referenceCode}/confirmation-history")]
```
Comment (kirillta) [inline Api/Controllers/AgentControllers/BookingsController.cs:346]: Please verify what we have the one term(`refcode`) across all routes

File: Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs [hunk 2]
```diff
@@ -161,6 +161,17 @@ public async Task<Result<List<BookingStatusHistoryEntry>>> GetBookingStatusHisto
...
 
 
+        public async Task<Result<List<BookingConfirmationHistoryEntry>>> GetBookingConfirmationHistory(string referenceCode)
+        {
+            var history = await _context.BookingConfirmationHistory
+                .Where(bch => bch.ReferenceCode == referenceCode)
+                .OrderBy(bch => bch.Id)
+                .ToListAsync();
+
+            return history ?? new(0);
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs:171]: `new(0)` is good, but one could add a static field for all empty returns.

File: Api/Controllers/AgentControllers/BookingsController.cs (hunk 1 above)
Comment (AlexeyShok) [reply Api/Controllers/AgentControllers/BookingsController.cs]: changed to reference-code

File: Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs (hunk 2 above)
Comment (AlexeyShok) [reply Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs]: Fixed

File: Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs [hunk 3]
```diff
@@ -262,7 +262,8 @@ Task<AccommodationBookingInfo.BookingAgentInformation> GetAgentInformation(int a
             BookingStatuses.Created,
             BookingStatuses.Invalid
         };
-        
+        private static readonly List<BookingConfirmationHistoryEntry> bookingConfirmationHistoryEmpty = new(0);
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs:265]: I think `_emptyBookingConfirmationHistory` sounds more natural, but that's a minor

**Review by kirillta — APPROVED** [review]

File: Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs (hunk 3 above)
Comment (AlexeyShok) [reply Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs]: Renamed

**Pull request merged**